paws_collector.js

```diff
@@ -498,17 +499,20 @@ class PawsCollector extends AlAwsCollector {
            
             if (err) {
                 /**
-                 * MS api return the logs but if build payload size > 10MB, and split logs array the in two part.
+                 * MS api return the logs but if build payload size > 10MB or if ingest return status code 307;
+                 * Split data per 10K message batches.
                  * Call the processLog again to send Logmsgs to ingest. 
                  */
-                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
-
-                    const half = Math.floor(logs.length / 2);
-                    const indexArray = [
-                        { start: 0, stop: half },
-                        { start: half, stop: logs.length }
-                    ];
-
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1) || (err.statusCode && err.statusCode === 307)) {
+                    const indexArray = [];
+                    if (logs.length > MAX_LOG_SIZE) {
+                        const oneBatch = Math.ceil(logs.length / MAX_LOG_SIZE);
+                        for (let i = 0; i < oneBatch; i++) {
+                            indexArray.push({ start: MAX_LOG_SIZE * i, stop: MAX_LOG_SIZE * (i + 1) });
```

Comment:
I suggest we use `array.slice()`, like [here](https://stackoverflow.com/questions/8495687/split-array-into-chunks)

Reply:
Here i am just forming the array base on MAX_LOG_BATCH_SIZE.